src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (const value of this) {
+      if (this.indexOf(value) === this.length - 1) {
```

Comment:
why do you need this? just use for (let i = 0; ...)

Reply:
@ivanchuk-yurii I changed, but I didn't understand how this simplifies the solution. Can you give me more details please, what I was supposed to change in the code?